File: src/question/month08/day0807/ms_0807.cpp
#include <iostream>

using namespace std;

int main() {
	int data[10], min = 10000, max = 0, i = 0;
	bool check[10];

	while (i < 10) 
	{
		scanf_s("%d", &data[i]);

		if (0 < data[i] && data[i] < 100)
			check[i] = true;
		else if (100 <= data[i] && data[i] < 10000)
			check[i] = false;

		i++;
	}

	for (i = 0; i < 10; i++) 
	{
		if (check[i]) 
		{
			if (data[i] > max)
				max = data[i];
		}
		else 
		{
			if (data[i] < min)
				min = data[i];
		}

		if (data[i] < 1)
			max = 100;

		if (data[i] >= 10000)
			min = 100;
	}

	cout << max << ' ' << min << endl;

	return 0;
}

File: year2020/month08/day0807/ms_0807.cs
using System;

namespace algorithm_30
{
    class ms_0807
    {
        static void Main(string[] args)
        {
            string[] input = Console.ReadLine().Split(' ');

            if(input.Length != 10)
                return;

            bool[] check = new bool[10];

            for(int i = 0; i < 10; i++) {
                int temp = Convert.ToInt32(input[i]);

                if(0 < temp && temp < 100)
                    check[i] = true;
                else if(100 <= temp && temp < 10000)
                    check[i] = false;
            }

            int max = 0, min = 10000;

            for(int i = 0; i < 10; i++) {
                int temp = Convert.ToInt32(input[i]);

                if(check[i]) {
                    if(temp > max)
                        max = temp;
                } else {
                    if(temp < min)
                        min = temp;
                }

                if(temp < 1)
                    max = 100;
                else if(temp >= 10000)
                    min = 100;
            }

            Console.WriteLine($"{max} {min}");
        }
    }
}
